src/Evolution/Systems/Cce/Actions/ReceiveWorldtubeData.hpp

```diff
@@ -0,0 +1,79 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include <tuple>
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "DataStructures/Variables.hpp"
+#include "Parallel/ConstGlobalCache.hpp"
+#include "Parallel/Invoke.hpp"
+#include "Time/TimeStepId.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TMPL.hpp"
+
+namespace Cce {
+namespace Actions {
+
+/*!
+ * \ingroup Actions
+ * \brief Takes the boundary data needed to perform the CCE linear solves as
+ * arguments and puts them in the \ref DataBoxGroup, updating the
+ * `Cce::Tags::BoundaryTime` accordingly.
+ *
+ * \details The boundary data is computed by a separate component, and packaged
+ * into a `Variables<tmpl::list<BoundaryTags...>>` which is sent in the argument
+ * of the simple action invocation. The `TimeStepId` is also provided to confirm
+ * the time associated with the passed boundary data.
+ *
+ * \note this action re-activates the algorithm by `perform_algorithm()` in case
```

Comment:
don't call `perform_algorithm` within an action; Wanting to do so means that this action should not be a simple_action.  Just promote this to a iterable action by taking the `is_ready` check from `BlockUntilBoundaryDataReceived` and get rid of `BlockUntilBoundaryDataReceived`

Reply:
Sorry, I need more information to work out what to do on this comment, because this simple action performs the data transfer via the arguments. Are you suggesting that I use the `receive_data` and inbox mechanisms instead of using the simple action, and examine the inboxes to determine is_ready?

Reply:
yes, that is what we are saying...